templates/slim/section.html.slim

```diff
@@ -1,12 +1,48 @@
+/ OPTIONS PROCESSING
 / hide slides on %conceal, %notitle and named "!"
 - titleless = (title = self.title) == '!'
 - hide_title = (titleless || (option? :notitle) || (option? :conceal))
+
 - vertical_slides = find_by(context: :section) {|section| section.level == 2 }
 
+/ extracting block image attributes to find an image to use as a 
+/ background_image attribute
+- data_background_image, data_background_size, data_background_repeat,
+  data_background_transition = nil
+
+- slide_images = find_by(:context => :image) { |image|
+-   image.attr(1) == 'background' || image.attr(1) == 'canvas'
+- }
+- unless slide_images.nil? or slide_images.length == 0
+  - slide_images.each do |image|
+    - data_background_image = image_uri(image.attr 'target')
+    / make sure no crash on nil and default values make sense
+    - data_background_size = image.attr 'size'
+    - data_background_repeat = image.attr 'repeat'
+    - data_background_transition = image.attr 'transition'
+    - break
+
+/ background-image section attribute overrides the image one
+- if attr? 'background-image'
+  - data_background_image = image_uri(attr 'background-image')
+
+- if attr? 'background-color'
+  - data_background_color = attr 'background-color'
+
+/ RENDERING
 / render parent section of vertical slides set
 - if @level == 1 && !vertical_slides.empty?
   section
-    section id=(titleless ? nil : @id) data-transition=(attr 'data-transition') data-transition-speed=(attr 'data-transition-speed') data-background=(attr 'data-background') data-background-size=(attr 'data-background-size') data-background-repeat=(attr 'data-background-repeat') data-background-transition=(attr 'data-background-transition')
+    / TODO: try to get rid of duplication w/ standalone slide section
+    section(id=(titleless ? nil : @id)
+      class=[@style,role]
```

Comment:
I don't think `@style` is really needed there. The style of a section is mostly reserved for special sections in DocBook. You can simply write this as:

```
class=roles
```